### PR TITLE
Improve sample id validation and enforce patient id validation

### PR DESCRIPTION
**Pull request by oplantalech**
Fix cbioportal/datahub#192 .

### CODE REVIEW

File: docs/File-Formats.md [hunk 1]
```diff
@@ -206,8 +206,8 @@ PATIENT_ID_2<TAB>LIVING<TAB>63.01<TAB>DiseaseFree<TAB>63.01<TAB>...
 
 ##### The samples file
 The file containing the sample attributes has two **required** columns:
-- **PATIENT_ID (required)**: A patient ID.
-- **SAMPLE_ID (required)**: A sample ID.
+- **PATIENT_ID (required)**: A patient ID. This field allows only numbers, letters, points, underscores and hyphens.
+- **SAMPLE_ID (required)**: A sample ID. This field allows only numbers, letters, points, underscores and hyphens.
```
Comment (sandertan) [inline docs/File-Formats.md]: Can you change these to "This field can only contain numbers, letters ......"?

File: core/src/main/scripts/importer/validateData.py [hunk 2]
```diff
@@ -2043,6 +2045,13 @@ def checkLine(self, data):
                         extra={'line_number': self.line_number,
                                'column_number': col_index + 1,
                                'cause': value})
+                if re.findall(self.INVALID_SAMPLE_ID_CHARACTERS, value):
+                    self.logger.error(
+                        'PATIENT_ID can only contain letters, numbers, points, '
+                        'underscores and hyphens',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
```
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: As the validation for PATIENT_ID is the same as for SAMPLE_ID, could you move this check to the super class `ClinicalValidator`?

Comment (sandertan) [reply core/src/main/scripts/importer/validateData.py]: Also, the "White space in PATIENT_ID is not supported" error is now captured by the new check. Can we remove this redundant check for both PATIENT_ID and SAMPLE_ID?

**Comment by sandertan** [comment]
Nice. Let's squash & merge.

File: core/src/main/scripts/importer/validateData.py [hunk 3]
```diff
@@ -1906,6 +1908,14 @@ def checkLine(self, data):
                         extra={'line_number': self.line_number,
                                'column_number': col_index + 1,
                                'cause': value})
+            if col_name == 'PATIENT_ID' or col_name == 'SAMPLE_ID':
+                if re.findall(self.INVALID_ID_CHARACTERS, value):
+                    self.logger.error(
+                        'PATIENT_ID and SAMPLE_ID can only contain letters, '
+                        'numbers, points, underscores and hyphens',
```
Comment (pieterlukasse) [inline core/src/main/scripts/importer/validateData.py]: replace `and hyphens` by `or hyphens`?

**Comment by oplantalech** [comment]
@jjgao Can you merge?

**Pull request merged**